packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -947,7 +947,12 @@ export class ConnectionManager implements IConnectionManager {
     }
 
     private readonly opHandler = (documentId: string, messagesArg: ISequencedDocumentMessage[]) => {
-        const messages = Array.isArray(messagesArg) ? messagesArg : [messagesArg];
+        let messages = Array.isArray(messagesArg) ? messagesArg : [messagesArg];
+        // TODO: might call DeltaConnection here to add the content back into a message
+        if (this.connection?.processInboundMessages) {
```

Comment:
I do not think this layer should know anything about it (assuming we are tasking driver to do all the heavy lifting - there are other options, including putting it all into this layer, or into container runtime layer).

I think you are probably thinking about ops potentially coming from multiple sources (storage, relay service).
But all ops from storage will come with content. Only ops on the same connection have a chance to not have content, as relay service can only sent "light" ops if it knows it's same client, i.e. clientId of sender and receiver are the same - i.e. the scope is single connection. Front-end (alfred) can do this kind of optimization only, and only in context of a single client connection (i.e., it may strip content only right befor sending op on the socket)

So I think all the logic can be encapsulated within DocumentDelteaConnection layer.